tests/multi_ptr/multi_ptr_comparison_op.h

```diff
@@ -0,0 +1,591 @@
+      // Expected value that will be returned after "value_mptr == nullptr"
+      // operator will be called
+      expected_results.value_mptr_nullptr_result_val = false;
+
+      const auto run_test_action = [](auto arr_acc, auto result_acc) {
+        multi_ptr_t nullptr_mptr(nullptr);
+        multi_ptr_t value_mptr(arr_acc);
+
+        auto &test_result = result_acc[0];
+
+        test_result.nullptr_nullptr_mptr_result_val = nullptr_mptr == nullptr;
+        test_result.nullptr_mptr_nullptr_result_val = nullptr == nullptr_mptr;
+        test_result.nullptr_value_mptr_result_val = value_mptr == nullptr;
+        test_result.value_mptr_nullptr_result_val = nullptr == value_mptr;
+      };
+
+      run_test(queue, run_test_action, expected_results);
+    }
+    SECTION(section_name(
+                "Check multi_ptr operator!=(std::nullptr_t, const multi_ptr&)")
+                .with("T", type_name)
+                .with("address_space", address_space_name)
+                .with("decorated", is_decorated_name)
+                .create()) {
+      // Variable that contains all variables that will be used to verify test
+      // result
+      detail::mptr_nullptr_mptr_test_results<T> expected_results;
+      // Expected value that will be returned after "nullptr != nullptr_mptr"
+      // operator will be called
+      expected_results.nullptr_nullptr_mptr_result_val = false;
+      // Expected value that will be returned after "nullptr_mptr != nullptr"
+      // operator will be called
+      expected_results.nullptr_mptr_nullptr_result_val = false;
+      // Expected value that will be returned after "nullptr != value_mptr"
+      // operator will be called
+      expected_results.nullptr_value_mptr_result_val = true;
+      // Expected value that will be returned after "value_mptr != nullptr"
+      // operator will be called
+      expected_results.value_mptr_nullptr_result_val = true;
+
+      const auto run_test_action = [](auto arr_acc, auto result_acc) {
+        multi_ptr_t nullptr_mptr(nullptr);
+        multi_ptr_t value_mptr(arr_acc);
+
+        auto &test_result = result_acc[0];
+
+        test_result.nullptr_nullptr_mptr_result_val = nullptr != nullptr_mptr;
+        test_result.nullptr_mptr_nullptr_result_val = nullptr_mptr != nullptr;
+        test_result.nullptr_value_mptr_result_val = nullptr != value_mptr;
+        test_result.value_mptr_nullptr_result_val = value_mptr != nullptr;
+      };
+
+      run_test(queue, run_test_action, expected_results);
+    }
+    SECTION(section_name(
+                "Check multi_ptr operator<(std::nullptr_t, const multi_ptr&)")
+                .with("T", type_name)
+                .with("address_space", address_space_name)
+                .with("decorated", is_decorated_name)
+                .create()) {
+      // Variable that contains all variables that will be used to verify test
+      // result. We expect that all of them will be true.
```

Comment:
The code above explicitly initialized true values anyway, which makes it too verbose IMHO. :)

Reply:
I'm going to mark this explicitly :)